contracts/contracts/oracle/OETHOracleUpdater.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { IOracleReceiver } from "./IOracleReceiver.sol";
+import { IVault } from "../interfaces/IVault.sol";
+import { AggregatorV3Interface } from "../interfaces/chainlink/AggregatorV3Interface.sol";
+import { Governable } from "../governance/Governable.sol";
+import { ICurvePool } from "../strategies/ICurvePool.sol";
+
+/**
+ * @title OETH Oracle Updater
+ * @notice Gathers on-chain OETH pricing data and updates the OETHOracle contract.
+ * @author Origin Protocol Inc
+ */
+contract OETHOracleUpdater is Governable {
+    /// @notice Max OETH price when redeeming via the vault to 18 decimals.
+    /// The vault charges a 0.5% withdraw fee and the oracle prices of
+    /// the vault collateral assets are capped at 1 so the max price is 0.995.
+    /// @dev A new OETHOracleUpdater needs to be deployed if the vault withdraw fee changes.
+    uint256 public constant MAX_VAULT_PRICE = 995e15;
+
+    /// @notice The OETH/ETH Curve pool
+    ICurvePool public immutable curvePool;
+    /// @notice The OETH Vault
+    IVault public immutable vault;
+
+    struct OracleUpdaterConfig {
+        address vault;
+        address curvePool;
+    }
+
+    event AddPrice(uint256 answer, uint256 vaultPrice, uint256 marketPrice);
+
+    /**
+     * @param _vault Address of the OETH Vault
+     * @param _curvePool Address of the OETH/ETH Curve pool
+     */
+    constructor(address _vault, address _curvePool) Governable() {
+        curvePool = ICurvePool(_curvePool);
+        vault = IVault(_vault);
+    }
+
+    /// @notice Adds a new on-chain, aggregated OETH/ETH price to 18 decimals to the specified Oracle.
+    /// @dev Callable by anyone as the prices are sourced and aggregated on-chain.
+    /// @param oracle Address of the Oracle that has authorized this contract to add prices.
+    function addPrice(IOracleReceiver oracle) external {
+        (
+            uint256 answer,
+            uint256 vaultPrice,
+            uint256 marketPrice
+        ) = _getPrices();
+
+        emit AddPrice(answer, vaultPrice, marketPrice);
```

Comment:
This event fires regardless of which oracle was updated. Someone could call a different success only oracle contract and confuse our analytics about when our oracle has been updated. We should probably include which oracle was updated in the event.

Reply:
excellent point. I'll add an indexed oracle address to the AddPrice event.